### PR TITLE
mgmt, support begin operation with syncpoll

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
Currently, try replace `Deployment.beginCreate` and `ResourceGroups.beginDelete` with this approach.

Note that there is case which is much more complicated than this, those with dependency resources need to be created before LRO, and following-up operation after LRO.

### CODE REVIEW

File: sdk/resources/mgmt/src/test/java/com/azure/resourcemanager/resources/DeploymentsTests.java [hunk 1]
```diff
@@ -224,4 +229,29 @@ public void canUpdateVirtualNetworkDeployment() throws Exception {
...
+            .withParametersLink(parametersUri, contentVersion)
+            .withMode(DeploymentMode.COMPLETE)
+            .beginCreate();
+        Deployment createdDeployment = acceptedDeployment.getAcceptedResult();
+        Assertions.assertNotEquals("Succeeded", createdDeployment.provisioningState());
+        PollResponse<Void> pollResponse = acceptedDeployment.getSyncPoller().poll();
+        while (pollResponse.getStatus() != LongRunningOperationStatus.SUCCESSFULLY_COMPLETED) {
+            SdkContext.sleep(1000);
+            pollResponse = acceptedDeployment.getSyncPoller().poll();
+        }
```
Comment (weidongxu-microsoft) [inline sdk/resources/mgmt/src/test/java/com/azure/resourcemanager/resources/DeploymentsTests.java:251]: Support polling by user.

File: sdk/resources/mgmt/src/main/java/com/azure/resourcemanager/resources/fluentcore/model/Accepted.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+     * @return the sync poller.
+     */
+    SyncPoller<Void, T> getSyncPoller();
+
+    /**
+     * Gets the final result of LRO.
+     *
+     * @return the final result.
+     */
+    T getFinalResult();
```
Comment (weidongxu-microsoft) [inline sdk/resources/mgmt/src/main/java/com/azure/resourcemanager/resources/fluentcore/model/Accepted.java:34]: Plan is to make this call return the resource, after LRO and after all pending tasks done.

File: sdk/resources/mgmt/src/main/java/com/azure/resourcemanager/resources/fluentcore/model/implementation/AcceptedImpl.java [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        private final Function<InnerT, T> wrapOperation;
+        private T finalResult;
+
+        SyncPollerImpl(SyncPoller<PollResult<InnerT>, InnerT> syncPoller, Function<InnerT, T> wrapOperation) {
+            this.syncPoller = syncPoller;
+            this.wrapOperation = wrapOperation;
+        }
+
+        @Override
+        public PollResponse<Void> poll() {
```
Comment (weidongxu-microsoft) [inline sdk/resources/mgmt/src/main/java/com/azure/resourcemanager/resources/fluentcore/model/implementation/AcceptedImpl.java:120]: Next plan is to give a concrete class for this current `Void`, to provide more information, e.g. poll url, response body, etc.

But that would require more changes in core-management, so it would likely be delayed to another PR.

File: sdk/compute/mgmt/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineImpl.java [hunk 4]
```diff
@@ -184,6 +191,7 @@
         super(name, innerModel, computeManager);
         this.storageManager = storageManager;
         this.networkManager = networkManager;
+        this.authorizationManager = authorizationManager;
```
Comment (ChenTanyi) [inline sdk/compute/mgmt/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineImpl.java:194]: Is there any usage of the `authorizationManager`?

Comment (weidongxu-microsoft) [reply sdk/compute/mgmt/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineImpl.java:194]: To create a new VirtualMachine, after polling is complete (line 1737)

```
                inner -> new VirtualMachineImpl(inner.name(), inner, this.manager(),
                    this.storageManager, this.networkManager, this.authorizationManager));
```

Comment (ChenTanyi) [reply sdk/compute/mgmt/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineImpl.java:194]: ok, I found it was used in a private resource
